### PR TITLE
chore: update source select dropdown, and api calls [f-7]

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR does two things:

- it adds `type` to the sources API call to ensure we can render it in the dropdown (and moves `custom_domains` to the right place where it exists in the API data returned)
- update the source select dropdown to bring it up to design standard.

Fun extra part: I added key-based navigation to the source select dropdown as well! 

https://user-images.githubusercontent.com/615334/144582278-4b5d087a-58d1-47fb-8f77-841e30e14d78.mov

**Source with a really long name**
image

 

### Stacked PR Chain: f-7
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#124|chore: fix layout and styles of source select and search bar |**Approved**|-|
|#125|feat(search): use hooks to style on input focus and show recent queries |**Approved**|#124|
|#126|chore: fix className overriding |**Approved**|#125|
|#127|chore: layout top bar correctly |**Approved**|#126|
|#128|chore: fix left icon  |**Approved**|#127|
|#129|👉 chore: update source select dropdown, and api calls |**Approved**|#128|
|#131|chore: fix layout and styles of source select and search bar |**Approved**|#129|
|#132|~~chore: have search-bar take up container width ~~|**Merged**|#131|
|#133|~~chore(search): refactor on-focus styling ~~|**Merged**|#132|
|#135|~~chore(search): add buttons to expander ~~|**Merged**|#133|
|#136|~~fix: use the active prop ~~|**Merged**|#135|
|#137|~~chore(search): refactor handlers ~~|**Merged**|#136|
|#138|~~fix: store current focus in variable ~~|**Merged**|#137|

### CODE REVIEW

**Comment by luqven** [comment]
This is 🔥 lit. Keyboard nav is ':chefs_kiss:' Will review soon.

File: frontend/src/common/hooks/useListSelectionBehavior.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        break;
+      case "Enter":
+        onSelectCurrentItem &&
+          selectedItemUnsafe &&
+          onSelectCurrentItem(selectedItemUnsafe);
+        break;
+    }
+  };
+
+  // Handle mouse interactions up and down to select items
```
Comment (luqven) [inline frontend/src/common/hooks/useListSelectionBehavior.tsx:76]:
```suggestion
  // Handle keyboard interactions up and down to select items
```

File: frontend/src/common/hooks/useListSelectionBehavior.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  return {
+    /**
+     * The currently selected item
+     */
+    currentSelected,
+    /**
+     * Should be called if another item should be highlighted/selected from another UI interaction, e.g. a mouse hover
+     */
+    handleOtherInteraction,
```
Comment (luqven) [inline frontend/src/common/hooks/useListSelectionBehavior.tsx:102]: Nice touch

File: frontend/src/components/buttons/Button.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export const Button = ({
       break;
   }
   return (
-    <button
+    <div
```
Comment (luqven) [inline frontend/src/components/buttons/Button.tsx:35]: What's the reason behind the change in the tag? I'm assuming it's to avoid event defaults?

File: frontend/src/components/buttons/dropdowns/SourceSelect.module.scss [hunk 4]
```diff
@@ -53,3 +54,56 @@
...
+  height: 12px;
+  margin-top: 5px;
+  content: "";
+  flex: 0 0 auto;
+  border-radius: 6px;
+  background: #19bf7c;
+
+  margin-right: 8px;
+}
+
```
Comment (luqven) [inline frontend/src/components/buttons/dropdowns/SourceSelect.module.scss:85]: Nice

File: frontend/src/components/buttons/dropdowns/SourceSelect.tsx [hunk 5]
```diff
@@ -11,40 +14,74 @@ interface Props {
...
-    setIsOpen(false);
+    setIsVisible(false);
     handleSelect(sourceId);
   };
 
+  const { visibleRef, setIsVisible, isVisible } = useClickOutside(false);
+
+  const { currentSelected, handleOtherInteraction } = useListSelectionBehavior({
+    items: sources,
+    active: isVisible,
```
Comment (luqven) [inline frontend/src/components/buttons/dropdowns/SourceSelect.tsx:40]: Nice

File: frontend/src/services/imgixAPIService.ts [hunk 6]
```diff
@@ -52,7 +52,7 @@ export const imgixAPI = {
      */
     async get(apiKey: string) {
       return await makeRequest<ImgixGETSourcesData>({
-        url: `sources?sort=name&fields[sources]=name&filter[enabled]=true`,
+        url: `sources?sort=name&fields[sources]=name,deployment.custom_domains,deployment.type&filter[enabled]=true`,
```
Comment (luqven) [inline frontend/src/services/imgixAPIService.ts:55]: qq. Does this filter sources that are deployed but have no assets?

If not we should tackle this as a subsequent PR. Maybe just where the "default" source is selected, so only a source with assets is set as "default" on the first render.

**Review by luqven — APPROVED** [review]
Great work 👏🏼 . Love this change.

Left some non-blocking qqs.

File: frontend/src/components/buttons/Button.tsx (hunk 3 above)
Comment (frederickfogerty) [reply frontend/src/components/buttons/Button.tsx:35]: It was causing a bug where if "Enter" was used to select a source, then it was triggering the button click as well. Making this a `div` fixed this issue

File: frontend/src/services/imgixAPIService.ts (hunk 6 above)
Comment (frederickfogerty) [reply frontend/src/services/imgixAPIService.ts:55]: It doesn't filter them out at this stage. 

Agreed that we should select a default source that has assets - we have a work item in the Phase 1 doc to implement this.
